# config/pure_pursuit.param.yaml
---

/**:
  ros__parameters:
    timeout_ms:                  20
    control_rate: 30.0
    delay_compensation_time: 0.17
    enable_overshoot_emergency: false
    enable_smooth_stop: true
    state_transition:
      drive_state_stop_dist: 0.3
      drive_state_offset_stop_dist: 1.0
      stopping_state_stop_dist: 0.49
      stopped_state_entry_vel: 0.1
      stopped_state_entry_acc: 0.1
      emergency_state_overshoot_stop_dist: 1.5
      emergency_state_traj_trans_dev: 3.0
      emergency_state_traj_rot_dev: 0.7
      emergency_acceleration: 0.5


    controller:
      minimum_lookahead_distance:      3.0
      maximum_lookahead_distance:      10.0
      speed_to_lookahead_ratio:        2.0
      is_interpolate_lookahead_point:  true
      is_delay_compensation:           false
      emergency_stop_distance:         0.1
      speed_thres_traveling_direction: 0.3
      maximum_acceleration:            1.0
      dist_front_rear_wheels:          2.7

    # smooth stop state
    smooth_stop_max_strong_acc: -0.5
    smooth_stop_min_strong_acc: -1.0
    smooth_stop_weak_acc: -0.3
    smooth_stop_weak_stop_acc: -0.8
    smooth_stop_strong_stop_acc: -3.4
    smooth_stop_max_fast_vel: 0.5
    smooth_stop_min_running_vel: 0.01
    smooth_stop_min_running_acc: 0.01
    smooth_stop_weak_stop_time: 0.8
    smooth_stop_weak_stop_dist: -0.3
    smooth_stop_strong_stop_dist: -0.5

    # stopped state
    stopped_vel: 0.0
    stopped_acc: -1.0
    stopped_jerk: -2.0